### PR TITLE
Launch zcashd

### PR DESCRIPTION
**Pull request by Oscar-Pepper**
on top of #1 

launches zcashd and builds up basic impls for Zcashd struct. run `cargo nextest run launch_zcashd --nocapture` to show the launch and shutdown with no errors/panics

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,2 +1,208 @@
...
+
+pub(crate) mod config;
+pub mod error;
 pub mod network;
+pub(crate) mod utils;
+
+const ZCASHD_STDOUT_LOG: &str = "stdout.log";
+
+/// Struct associated with Zcashd process.
+pub struct Zcashd {
```
Comment (idky137) [inline src/lib.rs:16]: Should zcashd be in lib.rs or should lib just hold traits and impls be i their own mod? I'm happy for this to merge like this but not sure whether code would be cleaner if moved into separate files?

**Review by idky137 — APPROVED** [review]
Have approved but maybe trait impls should be moved into their own mods? This can be done in a separate PR so isn't blocking

File: src/lib.rs (hunk 1 above)
Comment (Oscar-Pepper) [reply src/lib.rs:16]: i am not sure I like the structs and impls to be in different locations but i do plan to move these structs (and impls) to modules when it more clear what these modules should be, possibly `validator.rs` and `indexer.rs`

Comment (idky137) [reply src/lib.rs:16]: Ye I don't think separating everything necessary just that as this Repo grows lib.rs may start to get big. Something like that sound good!

**Comment by idky137** [comment]
Work towards Task 1 of [https://github.com/zingolabs/zaino/issues/51].

**Pull request merged**